sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/FileStream.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.batch.protocol;
+
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+
+/**
+ * An object to iterate over lines in a file.
+ * A LineStream is opened upon creation and is closed by invoking the close method.
+ *
+ * Inspired by {@link java.nio.file.DirectoryStream}
+ */
+public class FileStream implements Closeable, Iterable<String> {
+
+  private final File file;
+  private InputStream inputStream;
+
+  public FileStream(File file) {
+    this.file = file;
+  }
+
+  @Override
+  public Iterator<String> iterator() {
```

Comment:
But inputStream is erased so no more able to properly close it.
fileStream.iterator()
fileStream.iterator()
fileStream.close() -> close the second inputStream, the first is lost